src/metabase/related.clj

```diff
@@ -0,0 +1,249 @@
+(ns metabase.related
+  "Related entities recommendations."
+  (:require [clojure.set :as set]
+            [clojure.string :as str]
+            [medley.core :as m]
+            [metabase.api.common :as api]
+            [metabase.models
+             [card :refer [Card]]
+             [collection :refer [Collection]]
+             [dashboard :refer [Dashboard]]
+             [dashboard-card :refer [DashboardCard]]
+             [field :refer [Field]]
+             [interface :as mi]
+             [metric :refer [Metric]]
+             [query :refer [Query]]
+             [segment :refer [Segment]]
+             [table :refer [Table]]]
+            [schema.core :as s]
+            [toucan.db :as db]))
+
+(def ^:private ^Integer max-best-matches        3)
```

Comment:
I realized this a few weeks ago but integer literals in Clojure are usually `java.lang.Long`. So if you were to do `(class max-best-matches)` that's what you would get back. Since `Long` isn't a subclass of `Integer` (i.e. `Long`s are not `Integer`s) this tag is technically incorrect. We could either switch the tag to `^Long` or just do `(int 3)` instead of `3`. 

I doubt this will make a difference anywhere but it would be good to keep the type tags correct in case it does in some weird case

Reply:
Fixed. tnx